airbyte-server/src/main/java/io/airbyte/server/ConfigDumpExporter.java

```diff
@@ -145,4 +167,77 @@ private static Path buildConfigPath(final Path storageRoot, final String schemaT
         .resolve(String.format("%s.yaml", schemaType));
   }
 
+  public File exportWorkspace(UUID workspaceId) throws JsonValidationException, IOException, ConfigNotFoundException {
+    final Path tempFolder = Files.createTempDirectory(Path.of("/tmp"), ARCHIVE_FILE_NAME);
+    final File dump = Files.createTempFile(ARCHIVE_FILE_NAME, ".tar.gz").toFile();
+    exportVersionFile(tempFolder);
+    exportConfigsDatabase(tempFolder, workspaceId);
+
+    Archives.createArchive(tempFolder, dump.toPath());
+    return dump;
+  }
+
+  private void exportConfigsDatabase(Path parentFolder, UUID workspaceId) throws IOException, JsonValidationException, ConfigNotFoundException {
+    final List<SourceConnection> sourceConnections = configRepository.listSourceConnection()
+        .stream()
+        .filter(sourceConnection -> workspaceId.equals(sourceConnection.getWorkspaceId()))
+        .collect(Collectors.toList());
+    writeConfigsToArchive(parentFolder, ConfigSchema.SOURCE_CONNECTION.name(), sourceConnections.stream().map(Jsons::jsonNode));
+
+    final Map<UUID, StandardSourceDefinition> sourceDefinitionMap = new HashMap<>();
```

Comment:
this method is formidable. can we break it up a little? are there any DRY opportunities here?

Reply:
Yes, I just got the basic functionality working, I'll refactor parts and add tests now